core/src/main/java/com/graphhopper/routing/ev/NewCountry.java

```diff
@@ -0,0 +1,284 @@
+    OMN("Oman", "OM"),
+    PAK("Pakistan", "PK"),
+    PAN("Panama", "PA"),
+    PCN("Pitcairn Islands", "PN"),
+    PER("Peru", "PE"),
+    PHL("Philippines", "PH"),
+    PLW("Palau", "PW"),
+    PNG("Papua New Guinea", "PG"),
+    POL("Poland", "PL"),
+    PRK("North Korea", "KP"),
+    PRT("Portugal", "PT"),
+    PRY("Paraguay", "PY"),
+    PSE("Palestinian Territories", "PS"),
+    QAT("Qatar", "QA"),
+    ROU("Romania", "RO"),
+    RUS("Russia", "RU"),
+    RWA("Rwanda", "RW"),
+    SAU("Saudi Arabia", "SA"),
+    SDN("Sudan", "SD"),
+    SEN("Senegal", "SN"),
+    SGP("Singapore", "SG"),
+    SGS("South Georgia and the South Sandwich Islands", "GS"),
+    SHN("Saint Helena, Ascension and Tristan da Cunha", "SH"),
+    SLB("Solomon Islands", "SB"),
+    SLE("Sierra Leone", "SL"),
+    SLV("El Salvador", "SV"),
+    SMR("San Marino", "SM"),
+    SOM("Somalia", "SO"),
+    SRB("Serbia", "RS"),
+    SSD("South Sudan", "SS"),
+    STP("São Tomé and Príncipe", "ST"),
+    SUR("Suriname", "SR"),
+    SVK("Slovakia", "SK"),
+    SVN("Slovenia", "SI"),
+    SWE("Sweden", "SE"),
+    SWZ("Eswatini", "SZ"),
+    SYC("Seychelles", "SC"),
+    SYR("Syria", "SY"),
+    TCA("Turks and Caicos Islands", "TC"),
+    TCD("Chad", "TD"),
+    TGO("Togo", "TG"),
+    THA("Thailand", "TH"),
+    TJK("Tajikistan", "TJ"),
+    TKL("Tokelau", "TK"),
+    TKM("Turkmenistan", "TM"),
+    TLS("East Timor", "TL"),
+    TON("Tonga", "TO"),
+    TTO("Trinidad and Tobago", "TT"),
+    TUN("Tunisia", "TN"),
+    TUR("Turkey", "TR"),
+    TUV("Tuvalu", "TV"),
+    TWN("Taiwan", "TW"),
+    TZA("Tanzania", "TZ"),
+    UGA("Uganda", "UG"),
+    UKR("Ukraine", "UA"),
+    URY("Uruguay", "UY"),
+    USA("United States", "US"),
+    UZB("Uzbekistan", "UZ"),
+    VAT("Vatican City", "VA"),
+    VCT("Saint Vincent and the Grenadines", "VC"),
+    VEN("Venezuela", "VE"),
+    VGB("British Virgin Islands", "VG"),
+    VNM("Vietnam", "VN"),
+    VUT("Vanuatu", "VU"),
+    WSM("Samoa", "WS"),
+    XKX("Kosovo", "XK"),
+    YEM("Yemen", "YE"),
+    ZAF("South Africa", "ZA"),
+    ZMB("Zambia", "ZM"),
+    ZWE("Zimbabwe", "ZW");
+
+    public static final String KEY = "new_country";
+
+    private final String name;
+    private final String twoLetterCode;
+
+    NewCountry(String name, String twoLetterCode) {
+        this.name = name;
+        this.twoLetterCode = twoLetterCode;
+    }
+
+    /**
+     * @return the name:en field of this country
+     */
+    public String getName() {
+        return name;
+    }
+
+    /**
+     * @return the ISO3166-1:alpha2 code of this country
+     */
+    public String getTwoLetterCode() {
+        return twoLetterCode;
+    }
+
+    public static EnumEncodedValue<NewCountry> create() {
+        return new EnumEncodedValue<>(NewCountry.KEY, NewCountry.class);
+    }
+
+    public static NewCountry find(String name) {
+        if (name == null || name.isEmpty())
+            return MISSING;
+
+        for (NewCountry country : values()) {
```

Comment:
This is usually fast enough but this will be executed for each way in the OSMReader. Could we add a static HashMap and fill it once?

Reply:
I'll add a todo and then we should measure this before we add a hashmap. Maybe also calling `valueOf` and catching the exception so we can return `MISSING` if there is no match is fast.

Reply:
> Maybe also calling `valueOf` and catching the exception so we can return `MISSING` if there is no match is fast.

Instantiation of exceptions is quite costly. I'd avoid that in any hot code path.

Reply:
Do we even need `Country#find`? Currently we use `valueOf` in `OSMReader`, because the values are coming from the bundled countries.geojson file, so there is no need to handle the case where the value does not exist.